### PR TITLE
feat: Add created_at and updated_at columns to circle_integrations table

### PR DESCRIPTION
**Pull request by karelianpie**
## Motivation and Context

We don't have a way to track growth of integrations

## Description

Add `created_at` and `updated_at` columns to `circle_integrations` table

## Reviewers

@zashton

## Related Issue

Closes https://github.com/coordinape/coordinape/issues/1014

### CODE REVIEW

File: hasura/migrations/default/1657113614850_alter_table_public_circle_integrations_add_column_created_at/up.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+alter table "public"."circle_integrations" add column "created_at" timestamptz
```
Comment (zashton) [inline hasura/migrations/default/1657113614850_alter_table_public_circle_integrations_add_column_created_at/up.sql:1]:
```suggestion
alter table "public"."circle_integrations" add column "created_at" timestamptz NOT NULL DEFAULT now()
```

File: hasura/migrations/default/1657113629100_alter_table_public_circle_integrations_add_column_updated_at/up.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+alter table "public"."circle_integrations" add column "updated_at" timestamptz
```
Comment (zashton) [inline hasura/migrations/default/1657113629100_alter_table_public_circle_integrations_add_column_updated_at/up.sql:1]:
```suggestion
alter table "public"."circle_integrations" add column "updated_at" timestamptz NOT NULL DEFAULT now()
```

**Review by zashton — CHANGES_REQUESTED** [review]
Hi kraelianpie thanks for your contribution! 
just a couple of changes needed for the default values of the columns and you will need to add create DB trigger migration to make `updated_at` updated automatically when any other fields are changed (let me know if you need help with that)

**Comment by karelianpie** [comment]
> and you will need to add create DB trigger migration to make `updated_at` update automatically when any other fields are changed (let me know if you need help with that)

Thank you @zashton, could you just guide me on how to do that this one time

**Comment by zashton** [comment]
> > and you will need to add create DB trigger migration to make `updated_at` update automatically when any other fields are changed (let me know if you need help with that)
> 
> Thank you @zashton, could you just guide me on how to do that this one time
for the up migration
```sql
DO $$
BEGIN
    CREATE TRIGGER "set_circle_integrations_updated_at"
    BEFORE UPDATE ON "public"."circle_integrations"
    FOR EACH ROW
    EXECUTE PROCEDURE "public"."set_current_timestamp_updated_at"();
    COMMENT ON TRIGGER "set_circle_integrations_updated_at" ON "public"."circle_integrations"
    IS 'trigger to set value of column "updated_at" to current timestamp on row update';
EXCEPTION
    WHEN duplicate_object THEN
        RAISE NOTICE 'updated_at trigger already exists. Ignoring...';
END$$;
```

down migration
```sql
DROP TRIGGER IF EXISTS set_circle_integrations_updated_at on public."circle_integrations";
```

File: hasura/migrations/default/1657113614850_alter_table_public_circle_integrations_add_column_created_at/up.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+alter table "public"."circle_integrations" add column "created_at" timestamptz NOT NULL DEFAULT now()
+ null;
```
Comment (zashton) [inline hasura/migrations/default/1657113614850_alter_table_public_circle_integrations_add_column_created_at/up.sql:2]: you can remove the null here

Comment (karelianpie) [reply hasura/migrations/default/1657113614850_alter_table_public_circle_integrations_add_column_created_at/up.sql]: Removed with https://github.com/coordinape/coordinape/pull/1118/commits/4eddf3a51e1aa9a299b95f18b6ec701964ef7a4a, committed the GitHub suggestion and missed that

File: hasura/migrations/default/1657113629100_alter_table_public_circle_integrations_add_column_updated_at/up.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+alter table "public"."circle_integrations" add column "updated_at" timestamptz NOT NULL DEFAULT now()
+ null;
```
[the remark this answers is not in the document]
Comment (karelianpie) [reply hasura/migrations/default/1657113629100_alter_table_public_circle_integrations_add_column_updated_at/up.sql]: Same as above, removed with https://github.com/coordinape/coordinape/pull/1118/commits/4eddf3a51e1aa9a299b95f18b6ec701964ef7a4a

**Comment by karelianpie** [comment]
> > > and you will need to add create DB trigger migration to make `updated_at` update automatically when any other fields are changed (let me know if you need help with that)
> > 
> > 
> > Thank you @zashton, could you just guide me on how to do that this one time
> 
> for the up migration
> 
> ```sql
> DO $$
> BEGIN
>     CREATE TRIGGER "set_circle_integrations_updated_at"
>     BEFORE UPDATE ON "public"."circle_integrations"
>     FOR EACH ROW
>     EXECUTE PROCEDURE "public"."set_current_timestamp_updated_at"();
>     COMMENT ON TRIGGER "set_circle_integrations_updated_at" ON "public"."circle_integrations"
>     IS 'trigger to set value of column "updated_at" to current timestamp on row update';
> EXCEPTION
>     WHEN duplicate_object THEN
>         RAISE NOTICE 'updated_at trigger already exists. Ignoring...';
> END$$;
> ```
> 
> down migration
> 
> ```sql
> DROP TRIGGER IF EXISTS set_circle_integrations_updated_at on public."circle_integrations";
> ```

Thanks! Added with https://github.com/coordinape/coordinape/pull/1118/commits/f6b21bb8392501c27ae8a2c3b65bf6bd6e744543

**Review by zashton — APPROVED** [review]
thanks, looks good !

**Pull request merged**